okhttp-brotli/src/main/java/okhttp3/brotli/BrotliInterceptor.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.baulsupp.okurl.brotli
+
+import okhttp3.Interceptor
+import okhttp3.Response
+import okhttp3.ResponseBody
+import okio.buffer
+import okio.source
+import org.brotli.dec.BrotliInputStream
+
+/**
+ * Transparent Brotli response support.
+ *
+ * Adds Accept-Encoding: br to existing encodings and checks (and strips) for Content-Encoding: br in responses
+ */
+object BrotliInterceptor : Interceptor {
+  override fun intercept(chain: Interceptor.Chain): Response {
+    val request = chain.request().newBuilder().header("Accept-Encoding", "br").build()
```

Comment:
I would like okhttp to let me extend the "Accept-Encoding" header by "br" and not replace it completely to keep e.g. gzip compression in case the webserver does not support brotli yet.
Would that be possible?

Reply:
Let me confirm with a test. The intention was to add a new header value, not remove all others.

Reply:
@krombel thanks, good catch

Reply:
@swankjesse whats the right handling here?  add gzip also which BridgeInterceptor would otherwise?  Or check and only add if the client didn't define?  But still process if response is brotli?